src/test/java/blackjack/domain/card/DeckTest.java

```diff
@@ -0,0 +1,49 @@
+package blackjack.domain.card;
+
+import java.util.List;
+import java.util.Stack;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class DeckTest {
+
+    @Test
+    void 카드를_카드_셔플러로_섞을_수_있다() {
+        //given
+        Stack<Card> cards = new Stack<>();
+        cards.addAll(List.of(
+                new Card(Suit.CLUB, Rank.EIGHT),
+                new Card(Suit.CLUB, Rank.ONE),
+                new Card(Suit.CLUB, Rank.TEN)
+        ));
+        Deck deck = new Deck(cards);
+
+        //when
+        deck.shuffleCards(new FixCardsShuffler());
+
+        //then
+        Assertions.assertThat(deck.getCards()).isEqualTo(cards);
+    }
+
+    @Test
+    void 남아있는_카드가_없을_때_더_뽑을_수_없다() {
+        //given
+        Stack<Card> cards = new Stack<>();
+        cards.addAll(List.of(
+                new Card(Suit.CLUB, Rank.EIGHT),
+                new Card(Suit.CLUB, Rank.ONE),
+                new Card(Suit.CLUB, Rank.TEN)
+        ));
+        Deck deck = new Deck(cards);
+
+        //when
+        deck.draw();
+        deck.draw();
+        deck.draw();
+
+        //then
+        Assertions.assertThatThrownBy(() -> deck.draw())
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("남아있는 카드가 없습니다.");
```

Comment:
```suggestion
        Deck deck = new Deck(new Stack<>());

        assertThatThrownBy(() -> deck.draw())
                .isInstanceOf(IllegalArgumentException.class)
                .hasMessage("남아있는 카드가 없습니다.");
```
로도 충분히 테스트가능하지 않을까요?

Reply:
오! 그러네요

반영하였습니다!